conda_build/config.py

```diff
@@ -552,7 +552,9 @@ def compute_build_id(self, package_name, reset=False):
                 # important: this is recomputing prefixes and determines where work folders are.
                 self._build_id = build_id
                 if old_dir:
-                    shutil.move(old_dir, self.work_dir)
+                    work_dir = self.work_dir
+                    os.rmdir(work_dir)
```

Comment:
There's a lot of really annoying permissions stuff that might come up.  Using conda's rm_rf should help avoid any potential issues here:

```suggestion
                    rm_rf(work_dir)
```